lang/LangPrimSource/PyrArrayPrimitives.cpp

```diff
@@ -2087,6 +2089,96 @@ int prArrayContainsSeqColl(struct VMGlobals *g, int numArgsPushed)
 	return errNone;
 }
 
+int sc_arrayMaxDepth(PyrSlot *a, int depth)
+{
+	PyrSlot *slot, *endptr;
+	PyrObject *obj;
+	int size, newdepth;
+	newdepth = depth;
+	obj = slotRawObject(a);
+	size = obj->size;
+	slot = obj->slots - 1;
+	endptr = slot + size;
+	while (slot < endptr) {
+		++slot;
+		if (IsObj(slot)) {
```

Comment:
actually, looking through this I found it better not to use both `isKindOfSlot` and `isKindOf`. It seems to read better as it is now (I just followed the original implementation of `prArrayContainsSeqColl`.